internal/interpreter/api.rs

```diff
@@ -724,6 +770,19 @@ impl ComponentDefinition {
     }
 }
 
+/// CallbackDescriptor is a structure that's used to describe a callback declared in .slint
+/// code. It is returned from [`ComponentDefinition::callback_descriptors()`] and [`ComponentDefinition::global_callback_descriptors()`]
+#[non_exhaustive]
+#[derive(Debug, Clone)]
+pub struct CallbackDescriptor {
+    /// The name of the callback.
+    pub name: SharedString,
```

Comment:
Using a `SharedString` here instead of a `String` for consistency with the rest of the API. AFAICS only the properties and callbacks getters use `String`, while otherwise we use `SharedString`. I'd rather we transition to `SharedString`.

Reply:
I'm torn.  The API uses `SharedString` only for the "runtime" string which goes i properties, and String for everything else.  
But it is true it could have been a SharedString also for things in the component definition even if it is not shared.